recipe/meta.yaml

```diff
@@ -17,18 +17,21 @@ requirements:
     - setuptools
     - cython >=0.23
     - python
-    - six >=1.4
+    - six >=1.7.3
     - numpy x.x
-    - scipy >=0.9
+    - scipy >=0.17
+    - numpydoc >=0.6
   run:
     - python
-    - six >=1.4
-    - numpy x.x
```

Comment:
Yeah, I think people were trying to make the fix you proposed with `conda-build`. I'm unclear as to whether it will work with `conda-build-all`, which is used at staged-recipes and `conda-smithy` relies on, so have continued to encourage people to follow this conservative route and included it in both places.

That said, I totally agree `numpy x.x` is cryptic. It is also unnecessarily specific (only works with `numpy`). People have been discussing and working on fixing the handling of pinning dependencies in `conda-build` for a little over a year now. `conda-build` 3 (coming soon 😄) should finally fix this in a pretty general way. It may take a while for us to familiarize ourselves with how it works and integrate it properly at conda-forge, but it looks like we should be able to ease into it, which is good.

For more details (only if you are curious) please see the [docs]( http://stuarteberg.github.io/conda-docs/building/variants.html ). Note these are from a WIP PR of the docs that have since changed. ( https://github.com/conda/conda-docs/pull/414 ) The code changes are in PR ( https://github.com/conda/conda-build/pull/1585 ).